spring-batch-docs/asciidoc/common-patterns.adoc

```diff
@@ -98,12 +98,14 @@ in the following example:
 
 [source, java]
 ----
-public class PoisonPillItemWriter implements ItemWriter<T> {
+public class PoisonPillItemProcessor<T> implements ItemProcessor<T, T> {
 
-    public void write(T item) throws Exception {
```

Comment:
This should be "write(List<T> items)" and hence requires a loop in the method's body. I used an item processor instead.